### PR TITLE
Create travel api

### PR DESCRIPTION
**Pull request by logan-r**
### Tickets:

-   HCK-

### List of changes:

Create a travel model and api. Model is of form
```
{
  _id: ObjectId // The id of the travel document
  hackerId: ObjectId // The id of the hacker that is traveling
  accountId: ObjectId // The id of the user that is traveling
  request: number // The amount of money the hacker is requesting for travel
  offer: number // The amount of money we are giving the hacker for travel
  status: string // The state of the hacker's travel request
}
```

Following api end points were also created:
- GET `/travel/:id` gets a travel with a certain id
- GET `/travel/self` gets travel associated with the logged in user
- GET `/travel/email/:email` gets the travel associated with the user of a given email
- POST `/travel` creates a new travel
- PATCH `/travel/status/:id` updates the status field of a given traveler
- PATCH `/travel/offer/:id` updates the amount of money we are offering a given traveler

## Type of change

Please delete options that are not relevant.

-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

### How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

-   [ ] Test A
-   [ ] Test B

**Test Configuration**:

**Firmware version:**
**Hardware:**
**Toolchain:**
**SDK:**

### Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings
-   [x] Listed change(s) in the Changelog
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] I have made corresponding changes to the documentation
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: middlewares/travel.middleware.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+ * @param {(err?)=>void} next
+ * @return {void}
+ * @description
+ * Creates travel document after making sure there is no other hacker with the same linked accountId or hackerId
+ */
+async function createTravel(req, res, next) {
+    const travelDetails = req.body.travelDetails;
+
+    const exists = await Services.Hacker.findByAccountId(
+        travelDetails.accountId
```
[the remark this answers is not in the document]
Comment (logan-r) [reply middlewares/travel.middleware.js]: True true

File: constants/routes.constant.js [hunk 2]
```diff
@@ -150,6 +150,41 @@ const hackerRoutes = {
...
+        uri: "/api/travel/email/" + Constants.ROLE_CATEGORIES.SELF
+    },
+    getAnyByEmail: {
+        requestType: Constants.REQUEST_TYPES.GET,
+        uri: "/api/travel/email/" + Constants.ROLE_CATEGORIES.ALL
+    },
+    post: {
+        requestType: Constants.REQUEST_TYPES.POST,
+        uri: "/api/travel/"
+    },
```
Comment (Ty-Won) [inline constants/routes.constant.js]: what is this post? Did you mean create travel?

File: controllers/travel.controller.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * @description
+ *      Change a travel's information based on the trave;'s mongoID specified in req.params.id.
+ *      The id is moved to req.body.id from req.params.id by validation.
+ *      Returns a 200 status for an updated travel.
+ *      The new information is located in req.body.
+ */
+function updatedTravel(req, res) {
+    return res.status(200).json({
+        message: Constants.Success.TRAVEL_UPDATE,
+        data: req.body
```
Comment (Ty-Won) [inline controllers/travel.controller.js]: Should this be json like the other portions or no?

File: controllers/travel.controller.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+"use strict";
+const Constants = {
+    Success: require("../constants/success.constant"),
+    Error: require("../constants/error.constant")
+};
+
+function okay(req, res) {
```
Comment (Ty-Won) [inline controllers/travel.controller.js]: nit: okay sounds a bit odd as a function name

File: middlewares/travel.middleware.js [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+ * @param {{body: {travelDetails: {status: String, offer: Number}}}} req
+ * @param {JSON} res
+ * @param {(err?)=>void} next
+ * @return {void}
+ * @description Adds default status and offer to travelDetails.
+ */
+function addDefaultStatusAndOffer(req, res, next) {
+    req.body.travelDetails.status = "None";
+    req.body.travelDetails.offer = 0;
+    return next();
```
Comment (Ty-Won) [inline middlewares/travel.middleware.js]: Can this be added in parseTravel? or is this used for a specific purpose?

File: controllers/travel.controller.js (hunk 4 above)
Comment (logan-r) [reply controllers/travel.controller.js]: Whoops, this function was just for debugging - I'll remove it

File: constants/routes.constant.js (hunk 2 above)
Comment (logan-r) [reply constants/routes.constant.js]: Yeah, it's for create - thought I was following the naming convention from account and hacker, would it be better to rename this from `post` to `createTravel`?

**Pull request merged**